aws_lambda_powertools/utilities/parameters/secrets.py

```diff
@@ -116,6 +118,69 @@ def _get_multiple(self, path: str, **sdk_options) -> Dict[str, str]:
         """
         raise NotImplementedError()
 
+    def set_secret(
+            self,
+            *, # force keyword arguments
+            name: str,
+            secret: Optional[str],
+            secret_binary: Optional[str],
```

Comment:
I suggest to put the * after the secret key, as

set(key,value) makes a lot of sense.

I also suggest to change secret to value

```suggestion
    def set_secret(
            self,
            name: str,
            value: Union[str, bytes],
             *, # force keyword arguments

```

Reply:
Once 'value' is wither bytes or str, you can check the type by yourself and occupy the correct parameter accordingly, having better UX, and less error prone than the boto3 sdk
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/secretsmanager/client/put_secret_value.html#:~:text=of%20the%20secret.-,You%20must%20include%20SecretBinary%20or%20SecretString%2C%20but%20not%20both.,-We%20recommend%20you